code/game/objects/items/tcg/tcg_machines.dm

```diff
@@ -218,3 +218,176 @@ GLOBAL_LIST_EMPTY(tcgcard_machine_radial_choices)
+	
+	///Reference to the display panel generated by this button.
+	var/obj/effect/decal/trading_card_panel/display_panel_ref
+	///Typepath of the display panel generated.
+	var/display_panel_type = /obj/effect/decal/trading_card_panel
+	///Where the panel will be spawned in relation to the button on the X axis.
+	var/panel_offset_x = 1
+	///Where the panel will be spawned in relation to the button on the Y axis.
+	var/panel_offset_y = 0
+
+GLOBAL_LIST_EMPTY(tcgcard_mana_bar_radial_choices)
+
+/obj/machinery/trading_card_button/Initialize(mapload)
+	. = ..()
+	display_panel_ref = new display_panel_type(locate(x + panel_offset_x, y + panel_offset_y, z))
+
+/obj/machinery/trading_card_button/Destroy()
+	qdel(display_panel_ref)
+	. = ..()
+
+/obj/machinery/trading_card_button/attack_hand(mob/user)
+	var/list/choices = setup_global()
+	if(!length(choices))
+		choices = setup_radial()
+	var/choice = show_radial_menu(user, src, choices, custom_check = CALLBACK(src, PROC_REF(check_menu), user), require_near = TRUE, tooltips = TRUE)
+	if(!check_menu(user))
+		return
+	handle_choice(choice, user)
+	display_panel_ref.update_icon(UPDATE_OVERLAYS)
+	add_fingerprint(user)
+	return ..()
+
+/obj/machinery/trading_card_button/proc/setup_global()
+	return GLOB.tcgcard_mana_bar_radial_choices
+
+/obj/machinery/trading_card_button/proc/setup_radial()
+	var/radial_choices 
+	radial_choices = GLOB.tcgcard_mana_bar_radial_choices = list(
+	"Set Mana" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_set_mana"),
+	"Set Mana Slots" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_set_slots"),
+	"Next Turn" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_next"),
+	)
+	return radial_choices
+
+/obj/machinery/trading_card_button/proc/handle_choice(choice, mob/user)
+	var/input_value
+	switch(choice)
+		if("Set Mana")
+			input_value = tgui_input_number(user, "Please input total mana", "Set mana", display_panel_ref.gems, display_panel_ref.gem_slots, 0)
+			if(input_value)
+				display_panel_ref.gems = input_value
+		if("Set Mana Slots")
+			input_value = tgui_input_number(user, "Please input total mana slots", "Set mana slots", display_panel_ref.gem_slots, 10, 1)
+			if(input_value)
+				display_panel_ref.gem_slots = input_value
+		if("Next Turn")
+			if (display_panel_ref.gem_slots <= 9)
+				display_panel_ref.gem_slots += 1
+			display_panel_ref.gems = display_panel_ref.gem_slots
+
+/obj/machinery/trading_card_button/proc/check_menu(mob/living/user)
+	if(!istype(user))
+		return FALSE
+	if(user.incapacitated() || !user.Adjacent(src))
+		return FALSE
+	return TRUE
+
+/obj/machinery/trading_card_button/health
+	name = "life control panel"
+	desc = "a set of buttons that lets you keep track of your life shards when playing Tactical Game Cards."
+	icon_state = "health_buttons"
+	display_panel_type = /obj/effect/decal/trading_card_panel/health
+	panel_offset_x = -1
+
+GLOBAL_LIST_EMPTY(tcgcard_health_bar_radial_choices)
```

Comment:
Autodoc this. 
Also you're using globals instead of statics so you can override in subtypes without having to deal with generating it everytime right?

Reply:
You could also make a static list filled with (type = list(whatever_your_global_list_was)) but this is probably fine as well

Reply:
I brought up why globals were used for radial menus normally and was informed they were "a bit faster"

Reply:
Fair nuff